app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,30 @@
+class SessionsController < ApplicationController
+
+
+  def create
+    auth = request.env['omniauth.auth']
+    @user = User.where(:uid => auth['uid'].to_s).first || User.create_new_user(auth)
+    reset_session
+    session[:user_id] = @user.id
+    session[:ticket] = auth[:credentials][:ticket]
+    session[:auth] = auth
+    session[:logged_in] = true
+    redirect_to root_url
+  end
+
+  def failure
+    session[:logged_in] = false
```

Comment:
Looks like this method is not being tested, so just add one more test for this to get it to 100% 👍